### ISSUE TITLE
[clang] Missing diagnostic when return a local vector as a span

### ISSUE THREAD

**Issue by usx95** [llvm/llvm-project#108463]
https://godbolt.org/z/zes3e9jfr

This is a new false negative

```cpp
#include<string>
#include <vector>

template <typename T>
struct [[gsl::Owner]] StatusOr {
  const T &value() [[clang::lifetimebound]];
};

template<typename T>
struct [[gsl::Pointer]] Span {
    Span(const std::vector<T> &V): V(V){}
    const std::vector<T> &V;
};

// Pointer<Owner<Pointer>>> from Owner<Pointer>
Span<int*> test9(std::vector<int*> V) {
  return V; // No diagnsotic
}
```
cc @hokein

**Comment by hokein** [comment]
This issue was caused by the 'Container<GSLPointer>' patch. The patch was reverted yesterday, but it seems the version of Clang used by Compiler Explorer is slightly behind. I’ve confirmed that the issue no longer exists after the revert. (will address this in the reland version).

**Comment by usx95** [comment]
I don't think so. Compiler explorer is only 10hr behind atm at https://github.com/llvm/llvm-project/commit/6d859c1712b539fb36e799bd7b1801f4acacc1ac.
I still see https://godbolt.org/z/MzW3Yaxb8 new false-negative.

**Comment by hokein** [comment]
I didn't reproduce it with my local build, and it doesn't seem to trigger on Clang (assertion trunk), 
https://godbolt.org/z/dxW6KrjTP (this is weird).

**Comment by hokein** [comment]
https://godbolt.org/z/j49xx9GEP shows the clang trunk works as expected. (For your above link https://godbolt.org/z/zes3e9jfr, changing the source file will make it work again, we probably need to refresh the cache of Compiler explorer).

**Comment by usx95** [comment]
Aah I see. Thanks for clarifying.
Then we can close this and make this work in https://github.com/llvm/llvm-project/pull/108344
